integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonLoadDataCommand.scala

```diff
@@ -567,14 +568,18 @@ case class CarbonLoadDataCommand(
           carbonLoadModel,
           sparkSession)
     }
-    Dataset.ofRows(
-      sparkSession,
+    val convertedPlan =
       CarbonReflectionUtils.getInsertIntoCommand(
         convertRelation,
         partition,
         query,
-        isOverwriteTable,
-        false))
+        false,
+        false)
+    if (isOverwriteTable && partition.nonEmpty && table.isHivePartitionTable) {
```

Comment:
isHivePartitionTable command check is not required as it is already done

Reply:
ok